### ISSUE TITLE
Use in browser with requirejs

### ISSUE THREAD

**Issue by deefeloper** [metafloor/bwip-js#115]
Hi,

can someone explain to me how this is usable with the requirejs optimizer?

I tried using the following config:
```
requirejs.optimize({
  paths: {
    'xhr-fonts': '../../../node_modules/bwip-js/lib/xhr-fonts',
    'bwip-js': '../../../node_modules/bwip-js/bwipjs',

    bwipjs: 'util/bwip'
  },
  shim: {
    'bwip-js': {
      exports: 'BWIPJS',
      deps: ['xhr-fonts']
    }
  }
});
```

and try to include it like:
```
define([
  'xhr-fonts',
  'bwip-js'
], function (xhrFonts, BWIPJS) {
  return new BWIPJS(browserFonts, false);
});
```

however, the xhr-fonts script actually tries to find the `<script>` tag that loaded it... which is not possible when everything is compiled.

the browser-fonts script uses module.exports, so that is also not something supported.

Is there any easy way to use this as-is in a requirejs optimized environment?

**Comment by metafloor** [comment]
Have you tried working with the browser-* modules?  They are designed to work in the browser using module loaders.  The stuff under `lib/` is meant only for the demo.

**Comment by deefeloper** [comment]
Yes, the browser-* variants use node's `module.export` which a browser will not understand (error in chrome is something along the lines of `module is not defined`)

**Comment by metafloor** [comment]
I don't use requirejs (just react) so not sure I can help you.  Can you use webpack to gather all the modules into a single download?  That is usually a successful approach when dealing with module inter-dependencies.

**Comment by deefeloper** [comment]
I use requirejs because I do not want my code transpiled and besides that switching to webpack at this stage would require more effort than is feasible.

requirejs basically does almost the same as webpack in this case except it doesn't transpile.

I think it should be easy to make bwip-js compatible with any loader used in javascript, like how jquery, bootstrap and alot of other libs do.

I can spend some time investigating how much effort it would take, but I don't really understand how bwipjs works and why it's structured like this, but that might be because you use webpack and react.

As an alternative I could give you the boilerplate for creating compatible code and you could see if you think it's worth it to spend time on it.

I think it would be worth it as the library itself is awesome and works like a charm

**Comment by h3rbg3** [comment]
Ran into this brick wall today 👎

**Comment by metafloor** [comment]
This will be fixed in the 2.0 release, planned for early November '19.

**Comment by metafloor** [comment]
The 2.0 release is pending for this weekend or next week, depending on my schedule.  A (should-be) RequireJS-compatible UMD is used.  I do not use RequireJS so will be interested to hear of success/fail with the new code.
